yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryExecutor.java

```diff
@@ -203,46 +204,53 @@ private RelationalPreparedStatement prepareContinuationStatement(@Nonnull Relati
         // Check if we need to force continuations
         if ((maxRows == null) && forceContinuations) {
-            s.setMaxRows(FORCED_MAX_ROWS);
-            Object result = executeStatement(s, queryString);
-            if (result instanceof RelationalResultSet) {
-                List<RelationalResultSet> results = new ArrayList<>();
-                @SuppressWarnings("PMD.CloseResource")
-                RelationalResultSet resultSet = (RelationalResultSet)result;
-                RelationalResultSetMetaData metadata = resultSet.getMetaData(); // The first metadata will be used for all
-                boolean hasResult = resultSet.next(); // Initialize result set value retrieval. Has only one row.
-                // Edge case: when there are no results at all, we don't want to store the result set at all,
-                // to create an empty aggregated result set
-                if (hasResult) {
-                    results.add(resultSet);
-                }
-                // Have continuations - keep running the query
-                Continuation continuation = resultSet.getContinuation();
-                while (!continuation.atEnd()) {
-                    try (var s2 = prepareContinuationStatement(connection, continuation, FORCED_MAX_ROWS)) {
-                        resultSet = (RelationalResultSet)executeStatement(s2, null);
-                        resultSet.next(); // Initialize result set value retrieval. Has only one row.
-                        continuation = resultSet.getContinuation();
-                        if (!continuation.atEnd()) {
-                            results.add(resultSet);
-                        }
+            return executeStatementWithForcedContinuations(s, queryString, connection);
+        } else {
+            if (maxRows != null) {
+                s.setMaxRows(maxRows);
+            }
+            // No change to behavior
+            return executeStatement(s, queryString);
+        }
+    }
+
+    private Object executeStatementWithForcedContinuations(final @Nonnull Statement s, final @Nullable String queryString, final RelationalConnection connection) throws SQLException {
+        s.setMaxRows(FORCED_MAX_ROWS);
+        Object result = executeStatement(s, queryString);
+        if (result instanceof RelationalResultSet) {
+            @SuppressWarnings("PMD.CloseResource")
+            RelationalResultSet resultSet = (RelationalResultSet)result;
+            List<RelationalResultSet> results = new ArrayList<>();
+            RelationalResultSetMetaData metadata = resultSet.getMetaData(); // The first metadata will be used for all
+
+            boolean hasResult = resultSet.next(); // Initialize result set value retrieval. Has only one row.
+            // Edge case: when there are no results at all, we don't want to store the result set at all,
+            // to create an empty aggregated result set
+            if (hasResult) {
```

Comment:
Should this return the `resultSet` if `!hasResult` ? Rather than returning an empty `AggregateResultSet`?

Reply:
Sure, that would work.